### ISSUE TITLE
MySQL password changes only take effect for logged out users

### ISSUE THREAD

**Issue by rdn4github** [MutonUfoAI/pgina#125]
I am using the MySQL plugin for authentication.  Password changes are made by an external application.  For logged-out users, this works fine.  But for logged-in users, the user must use the old password until the he or she logs out.

Example:  A user is trying to reconnect to a disconnected Remote Desktop session after a password change.  This fails with the new password even though the MySQL record has been properly modified.

I imagine this affects other Authentication Plugins as well (LDAP and/or RADIUS), but I don't use them, so I don't know for sure.

I realize this is by design, as the documented flow-chart shows that authentication for logged-in users do not go through the plugin chain.  But shouldn't the "Is User Logged In?" check be between the Authentication stage and the Authorization stage?  Then, if the user is logged in, the results of the Authentication stage would be returned upstream with the rest of the plugin chain ignored.

**Comment by MutonUfoAI** [comment]
http://mutonufoai.github.io/pgina/documentation/user.html#change_password
As you can see, MySQL doesn't support password change option.

**Comment by rdn4github** [comment]
True, which is why the password must be changed via an external application.  I understand that behavior and have no issue with it.  What I'm wondering is why the Authentication plugins (and _only_ the Authentication plugins) aren't fired when the user is already logged in.  (Or, as in the case of RDP, the user is simply disconnected but not logged out.)

Maybe I'm jumping to a particular solution when another one exists, so here is my problem, in a nutshell:

1) User connects to computer via RDP, User logs out, User changes password, User connects again to computer via RDP and the new password: **This WORKS**
2) User connects to computer via RDP, User disconnects, User changes password, User connects again to computer via RDP and the new password: **This FAILS** (Bad Password)

Any idea how to get #2 to succeed using the MysSQL plugin for Authentication?

Thanks

**Comment by MutonUfoAI** [comment]
http://mutonufoai.github.io/pgina/documentation/user.html#how_pgina_works
In an unlock scenario NO plugin in executed.
Why?
Because plugins can't distinguish between a login and an unlock.
